localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/api_key_validation.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from typing import Optional
+
+from localstack.aws.api.apigateway import ApiKey, ApiKeySourceType, RestApi
+from localstack.http import Response
+
+from ..api import RestApiGatewayHandler, RestApiGatewayHandlerChain
+from ..context import InvocationRequest, RestApiInvocationContext
+from ..gateway_response import InvalidAPIKeyError
+from ..moto_helpers import get_api_key, get_usage_plan_keys, get_usage_plans
+from ..variables import ContextVarsIdentity
+
+LOG = logging.getLogger(__name__)
+
+
+class ApiKeyValidationHandler(RestApiGatewayHandler):
+    """
+    Handles Api key validation.
+    If an api key is required, we will validate that a usage plan associated with that stage
+    has a usage plan key with the corresponding value.
+    """
+
+    # TODO We currently do not support rate limiting or quota limit. As such we are not raising any related Exception
+
+    def __call__(
+        self,
+        chain: RestApiGatewayHandlerChain,
+        context: RestApiInvocationContext,
+        response: Response,
+    ):
+        method = context.resource_method
+        request = context.invocation_request
+        rest_api = context.deployment.rest_api.rest_api
+
+        # If api key is not required by the method, we can exit the handler
+        if not method.get("apiKeyRequired"):
+            return
+
+        # If the Identity context was not created yet, instantiate it and attach it to the context variables
+        if not (identity := context.context_variables.get("identity", ContextVarsIdentity())):
+            context.context_variables["identity"] = identity
```

Comment:
You can simplify with `setdefault(key, default_value)` which does the same thing under the hood:
```suggestion
        identity = context.context_variables.setdefault("identity", ContextVarsIdentity())
```